Replaced "lazy computation" with "lazy expression"

WallaceKelly opened the pull request:
The term "lazy computation" makes it sound like the feature is based on computations, like async { } and query { }. The term "lazy expression" is more clear.

## Summary

Replaced "computation" with "expression"

Fixes #Issue_Number (if available)

WallaceKelly committed:
Replaced "lazy computation" with "lazy expression"

The term "lazy computation" makes it sound like the feature is based on computations, like async { } and query { }. The term "lazy expression" is more clear.

dnfclas commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=11284) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=11284) before we can accept your contribution.<br/><br/>:x: WallaceKelly [sign now](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=11284)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docs?pullRequest=11284) it.</sub>

WallaceKelly commented:
These two should also be edited:

- https://fsharp.org/specs/language-spec/4.1/FSharpSpec-4.1-latest.pdf
- https://msdn.microsoft.com/visualfsharpdocs/conceptual/control.lazyextensions-module-%5bfsharp%5d

I can take care of those if this change is approved.

WallaceKelly committed:
Renamed "Lazy Computations" to "Lazy Expressions"

WallaceKelly committed:
Renamed "Lazy Computations" to "Lazy Expressions"

WallaceKelly committed:
Renamed "Lazy Computations" to "Lazy Expressions"

WallaceKelly commented:
I received an email stating that renaming "lazy-compuations.md" to "lazy-expression.md" broke three links within the docs. I fixed those with three additional commits.

mairaw commented:
I’ll let @cartermp review the proposed change but file renames also need to include an entry in the following file:
https://github.com/dotnet/docs/blob/master/.openpublishing.redirection.json

WallaceKelly committed:
Redirect "Lazy Computations" to "Lazy Expressions"

This redirect supports an overall renaming of "lazy computations" to "lazy evaluations". I included `"redirect_document_id": true`, though I don't know what that means.

WallaceKelly commented:
@mairaw Thanks for the pointer to the redirect file. I added an entry for renaming "lazy computations" to "lazy expressions". I followed the pattern I saw in the other entries. I was not sure what `"redirect_document_id": true` means. It appears in some entries, but not others. I included it in my commit.

cartermp commented on docs/fsharp/language-reference/lazy-expressions.md:
```diff
@@ -1,11 +1,11 @@
-description: Learn how F# lazy computations can improve the performance of your apps and libraries.
+title: Lazy Expressions
+description: Learn how F# lazy expressions can improve the performance of your apps and libraries.
 ms.date: 05/16/2016
```
Can you modify the date to be today?

cartermp reviewed (changes requested):
LGTM, one minor change

WallaceKelly committed:
Updated ms,date on renamed lazy-expressions.md

cartermp reviewed (approved):
Thanks @WallaceKelly
